### DISCUSSION TITLE
[@types/plotly.js] Can the d3 import be removed from plotly.js?

### DISCUSSION THREAD

**Discussion by JonathanMcIntyre** [DefinitelyTyped/DefinitelyTyped#62305] [Issues with a @types package]
Can the d3 import be removed from the following file?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/plotly.js/index.d.ts

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
Historically, the `plotly.js` ecosystem had used `d3` via import of `d3` and re-exported the symbol. 
This is no longer a case in current `plotly.js` definition. It has only an import which does nothing,
but still requiring a path property configured in `tsconfig.json`.
I beieve plotly is now using internal fork of d3 internally:
https://github.com/plotly/d3/blob/master/package.json#L2
but this is not required to be part of public api in `plotly.js` types. Am I correct folks?

**Reply by brammitch** [reply]
I believe you're correct.

See https://community.plotly.com/t/plotly-js-2-0-is-coming-soon/50055

Under the heading, "What is getting changed or removed":

> The Plotly.d3 object: this object currently mirrors all of version 3 of the D3 library. Version 4 of D3 came out over 4 years ago (the current latest version is 6.5!) and the state of Javascript packaging has evolved such that it’s now much easier for Javascript developers to “bring their own” version of d3 to their app rather than rely on the one provided by Plotly.js.

**Reply by peterblazejewicz** [reply]
@brammitch thanks! I'll create PR to update the related definitions.
